modules/commons/src/main/java/org/apache/synapse/commons/json/JsonUtil.java

```diff
@@ -367,16 +381,49 @@ public static XMLStreamReader getReader(InputStream jsonStream) throws XMLStream
      * @throws AxisFault
      */
     public static void writeAsJson(OMElement element, OutputStream outputStream) throws AxisFault {
-        XMLEventReader xmlEventReader = null;
-        XMLEventWriter jsonWriter = null;
         if (element == null) {
-            logger.error("#writeAsJson. OMElement is null. Cannot convert to JSON.");
             throw new AxisFault("OMElement is null. Cannot convert to JSON.");
         }
         if (outputStream == null) {
             return;
         }
         transformElement(element, true);
+        convertOMElementToJson(element, outputStream);
+    }
+
+    /**
+     * Converts an XML element to its JSON representation and writes it to an output stream.<br/>
+     * Note that this method removes all existing namespace declarations and namespace prefixes of the provided XML
+     * element<br/>
+     *
+     * @param requiredProperties Message context properties
+     * @param element            XML element of which JSON representation is expected.
+     * @param outputStream       Output Stream to write the JSON representation.<br/>
+     *                           At the end of a successful conversion, its flush method will be called.
+     * @throws AxisFault
+     */
+    public static void writeAsJson(Map requiredProperties, OMElement element, OutputStream outputStream) throws
```

Comment:
let's reorder function signature. Give priority to important arguments. There may be no standard convention for that but it's logical and improve readability 
Ex : 
```
public static void writeAsJson(OMElement element, OutputStream outputStream, Map requiredProperties)
```

Reply:
Done